src/Maui/Prism.Maui/Navigation/PageNavigationService.cs

```diff
@@ -333,6 +333,44 @@ public virtual async Task<INavigationResult> NavigateAsync(Uri uri, INavigationP
         }
     }
 
+    /// <inheritdoc />
+    public virtual async Task<INavigationResult> NavigateFromAsync(string viewName, Uri route, INavigationParameters parameters)
+    {
+        await WaitForPendingNavigationRequests();
+
+        try
+        {
+            parameters ??= new NavigationParameters();
+
+            NavigationSource = PageNavigationSource.NavigationService;
+
+            var routeSegments = UriParsingHelper.GetUriSegments(route);
+
+            // Find a page that matches the viewName.
+            var page = GetCurrentPage();
+            while (page != null)
+            {
+                if (page is not null && ViewModelLocator.GetNavigationName(page) == viewName)
+                    break;
+                page = page.GetParentPage();
```

Comment:
Parent page? Shouldn't this be the page before the current page in the stack and if it is the root page then we get the parent page?

Reply:
@AmrAlSayed0 is correct. While we do care about the parent in some scenarios, what we really need is the previous page in the navigation stack.

Reply:
There are still some issues remaining, but I have made corrections to use the NavigationStack as described in the following link.

https://github.com/PrismLibrary/Prism/blob/7f8a12fad2a1ad75ecb1a78041f58bf204311afb/src/Maui/Prism.Maui/Navigation/PageNavigationService.cs#L351-L383